sync/awsdatabricks.py

```diff
@@ -181,17 +192,23 @@ def get_cluster_report(
     :type compute_type: str
     :param allow_incomplete: Whether creating a cluster report with incomplete data should be allowable
     :type allow_incomplete: bool, optional, defaults to False
+    :param exclude_tasks: Keys of tasks (task names) to exclude from the report
+    :type exclude_tasks: Collection[str], optional, defaults to None
     :return: cluster report
     :rtype: Response[DatabricksClusterReport]
     """
     run = get_default_client().get_run(run_id)
     if "error_code" in run:
         return Response(error=DatabricksAPIError(**run))
 
-    if run["state"].get("result_state") != "SUCCESS":
-        return Response(error=DatabricksError(message="Run did not successfully complete"))
+    tasks = [
+        task for task in run["tasks"] if not exclude_tasks or task["task_key"] not in exclude_tasks
+    ]
 
-    cluster_id_response = _get_run_cluster_id(run["tasks"])
+    if any(task["state"].get("result_state") != "SUCCESS" for task in tasks):
+        return Response(error=DatabricksError(message="Tasks did not complete successfully"))
+
+    cluster_id_response = _get_run_cluster_id(tasks)
```

Comment:
Just to double-check my understanding here -

- This allows us to get "in-progress" runs because we are no longer looking at the `run` state itself, but instead the state of each of the tasks that are not in the exclusion list,
- Polling for the eventlog will be fine because we are expecting this task to run on some other compute resource, so the cluster the non-excluded tasks run on should terminate as expected
  - Should we enforce this uniqueness? This exclusion mechanism is more generic than a "special task" one, so requiring this may not quite fit if you want to exclude other tasks for any reason

Reply:
I'm just thinking though that if we _don't_ enforce this somehow, we could accidentally cause an "infinite wait", since the cluster that we are polling will be the same one that we're running on